autopatch/autopatch.go

```diff
@@ -400,3 +402,32 @@ func makeOptionalSchema(s *huma.Schema) *huma.Schema {
 
 	return optionalSchema
 }
+
+// This function help to find the relative path of the resource to patch
+// this allow to handle potential prefix in the path
+// for example if the requestPath is /api/v1/user/1 and the put path is /user/{id}
+// the function will return /user/1
+func findRelativeRessourcePath(requestPath string, putPath string) string {
+	putPathParts := strings.Split(putPath, "/")
+	// if the path is not deep enough, we return the original path
+	if len(putPathParts) < 2 {
```

Comment:
What determines this magic number here? Why `2`? Are there any cases where this could fail?

Reply:
The number 2 is here to handle this case:

if putPath is `/{id}` ->` len(putPathParts == 1` that mean that we don't have a ressource name to find,

A case that could be not covered by this could be:

GET {prefix}/singleRessource
PUT {prefix}/singleRessource

it also protect the table access `wantedPrefix := putPathParts[1]`

Reply:
In the case that is not supported we just fall back to the default behavior